### PR TITLE
Lower halving_* intrinsics without widening

### PR DESCRIPTION
**Pull request by vksnk**

### CODE REVIEW

**Comment by dsharletg** [comment]
I think halving_* and rounding_halving_* are now tricky enough that we should test for correctness here, like saturating_*: https://github.com/halide/Halide/blob/master/test/correctness/intrinsics.cpp#L30

**Comment by vksnk** [comment]
I've added a test, please take a look.

File: test/correctness/intrinsics.cpp [hunk 1]
```diff
@@ -37,25 +37,23 @@ void check_saturating_add_sub() {
...
-                std::cerr << "expected: " << reference_add << "\n";
-                abort();
-            }
-            if (!can_prove(result_sub == make_const(halide_t, reference_sub))) {
-                std::cerr << "failure!\n";
-                std::cerr << "test: " << sub << "\n";
-                std::cerr << "result: " << result_sub << "\n";
-                std::cerr << "expected: " << reference_sub << "\n";
-                abort();
+            std::vector<std::pair<Expr, int64_t>> intrinsics_with_reference_answer = {
```
Comment (dsharletg) [inline test/correctness/intrinsics.cpp:40]: Probably best to just use a regular C array here. Constructing/destroying the vector here might be pretty expensive. I had to tweak N to make the test run in a reasonable amount of time.

**Review by dsharletg — APPROVED** [review]

**Comment by vksnk** [comment]
Looks like failures are unrelated, merging in.

**Pull request merged**